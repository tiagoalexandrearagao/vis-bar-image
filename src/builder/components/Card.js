Object.defineProperty(exports, "__esModule", { value: true });
exports.getCard = void 0;
var Common_1 = require("./Common");
var getCard = function (card) {
  var CONFIG = getConfiguration(card);
  var cardHtml = [];
  cardHtml.push(
    " <div class='paper card "
      .concat(CONFIG.shadowClass, "' style='\n                          ")
      .concat((0, Common_1.getCommonProps)(card), " ")
      .concat(card.cardStyles || "", "; \n                          ")
      .concat(CONFIG.cardBorder, ";\n          '>  ")
  );
  if (card.heading) {
    cardHtml.push(
      "<div class='cardHead "
        .concat(CONFIG.headShadow, "' style= ' ")
        .concat(
          CONFIG.headBackgroundColor,
          ";\n                                                    "
        )
        .concat(
          CONFIG.headColor,
          ";\n                                                    "
        )
        .concat(
          CONFIG.headCenter,
          ";\n                                                    "
        )
        .concat(
          card.headStyles || "",
          ";\n                  '>\n                  "
        )
        .concat(card.heading || "", "\n              </div>")
    );
  }
  cardHtml.push(
    "  <div class='cardContents' style= '"
      .concat(
        CONFIG.contentCenter,
        "; \n                                                      "
      )
      .concat(
        card.contentStyles || "",
        ";\n                    '>\n                    "
      )
      .concat(card.content, "\n                </div>\n              </div>  ")
  );
  return cardHtml.join("");
};
exports.getCard = getCard;
var getConfiguration = function (card) {
  var HEAD_CENTER = card.isHeadingCenter
    ? "text-align: center"
    : "text-align: left";
  var CONTENT_CENTER = card.isContentCenter
    ? "text-align: center"
    : "text-align:left";
  var SHADOW_CLASS = card.hasShadow ? "shadow" : "";
  var HEAD_SHADOW_CLASS = card.hasHeadingShadow ? "shadow" : "";
  var CARD_BORDER =
    card.borderColor != null
      ? " border: 1px solid  ".concat(
          (0, Common_1.getThemeShades)(card.borderColor).l3
        )
      : "";
  var headBackgroundColor = "";
  var headColor = "";
  if (card.headBackgroundColor == undefined) {
    headBackgroundColor = "background: rgba(0,0,0,.03)";
    headColor = "color: rgba(0,0,0,.54)";
  } else {
    headBackgroundColor = "background: ".concat(
      (0, Common_1.getThemeShades)(card.headBackgroundColor).l3
    );
    headColor = "color: #fff";
  }
  var CONFIG = {
    headCenter: HEAD_CENTER,
    cardBorder: CARD_BORDER,
    contentCenter: CONTENT_CENTER,
    shadowClass: SHADOW_CLASS,
    headBackgroundColor: headBackgroundColor,
    headColor: headColor,
    headShadow: HEAD_SHADOW_CLASS,
  };
  return CONFIG;
};
