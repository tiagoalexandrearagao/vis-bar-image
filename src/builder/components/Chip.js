Object.defineProperty(exports, "__esModule", { value: true });
exports.getChip = void 0;
var getChip = function (chip) {
  return "<div class='chip'><span class='chipText'>".concat(
    chip.text,
    "</span></div>"
  );
};
exports.getChip = getChip;
