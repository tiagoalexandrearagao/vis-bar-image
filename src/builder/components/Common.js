Object.defineProperty(exports, "__esModule", { value: true });
exports.getCommonProps = exports.getThemeShades = exports.Colors = void 0;
var Colors;
(function (Colors) {
  Colors[(Colors["blue"] = 0)] = "blue";
  Colors[(Colors["pink"] = 1)] = "pink";
  Colors[(Colors["green"] = 2)] = "green";
  Colors[(Colors["orange"] = 3)] = "orange";
  Colors[(Colors["teal"] = 4)] = "teal";
  Colors[(Colors["indigo"] = 5)] = "indigo";
  Colors[(Colors["grey"] = 6)] = "grey";
})((Colors = exports.Colors || (exports.Colors = {})));
var getThemeShades = function (theme) {
  // l1 = 100, l2 = 300, l3 = 800
  switch (theme) {
    case Colors.blue:
      return {
        l1: "#BBDEFB",
        l2: "#64B5F6",
        l3: "#1565C0",
      };
    case Colors.pink:
      return {
        l1: "#F8BBD0",
        l2: "#F06292",
        l3: "#AD1457",
      };
    case Colors.green:
      return {
        l1: "#C8E6C9",
        l2: "#81C784",
        l3: "#2E7D32",
      };
    case Colors.orange:
      return {
        l1: "#FFE0B2",
        l2: "#FFB74D",
        l3: "#EF6C00",
      };
    case Colors.teal:
      return {
        l1: "#B2DFDB",
        l2: "#4DB6AC",
        l3: "#00695C",
      };
    case Colors.indigo:
      return {
        l1: "#C5CAE9",
        l2: "#7986CB",
        l3: "#283593",
      };
    case Colors.grey:
      return {
        l1: "#F5F5F5",
        l2: "#BDBDBD",
        l3: "#424242",
      };
    default:
      return {
        l1: "#F5F5F5",
        l2: "#BDBDBD",
        l3: "#424242",
      };
  }
};
exports.getThemeShades = getThemeShades;
//TODO use join instead of +=
var getCommonProps = function (ele) {
  var styles = [];
  styles.push("width: ".concat(ele.width || "fit-content", ";"));
  styles.push("height: ".concat(ele.height || "auto", ";"));
  styles.push("margin: 4px 4px;");
  return styles.join("");
};
exports.getCommonProps = getCommonProps;
