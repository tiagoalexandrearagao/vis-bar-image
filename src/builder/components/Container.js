Object.defineProperty(exports, "__esModule", { value: true });
exports.getContainer = exports.Position = void 0;
var Common_1 = require("./Common");
var Position;
(function (Position) {
  Position[(Position["topCenter"] = 0)] = "topCenter";
  Position[(Position["bottomCenter"] = 1)] = "bottomCenter";
  Position[(Position["center"] = 2)] = "center";
  Position[(Position["rightTop"] = 3)] = "rightTop";
  Position[(Position["rightCenter"] = 4)] = "rightCenter";
  Position[(Position["rightBottom"] = 5)] = "rightBottom";
  Position[(Position["leftTop"] = 6)] = "leftTop";
  Position[(Position["leftCenter"] = 7)] = "leftCenter";
  Position[(Position["leftBottom"] = 8)] = "leftBottom";
})((Position = exports.Position || (exports.Position = {})));
var getContainer = function (container) {
  var fAlignProps = "";
  var dAlignProps = "";
  var BACKGROUND_COLOR = container.backgroundColor
    ? (0, Common_1.getThemeShades)(container.backgroundColor).l1
    : "#fff";
  switch (container.alignItems) {
    case Position.leftTop:
      fAlignProps = "justify-content:flex-start;";
      dAlignProps = "margin-bottom: auto;";
      break;
    case Position.leftCenter:
      fAlignProps = "justify-content:flex-start;";
      dAlignProps += "margin-bottom: auto;";
      dAlignProps += "margin-top: auto;";
      break;
    case Position.leftBottom:
      fAlignProps = "justify-content:flex-start;";
      dAlignProps = "margin-top: auto;";
      break;
    case Position.rightTop:
      fAlignProps = "justify-content:flex-end;";
      dAlignProps = "margin-bottom: auto;";
      break;
    case Position.rightCenter:
      fAlignProps = "justify-content:flex-end;";
      dAlignProps += "margin-bottom: auto;";
      dAlignProps += "margin-top: auto;";
      break;
    case Position.rightBottom:
      fAlignProps = "justify-content:flex-end;";
      dAlignProps = "margin-top: auto;";
      break;
    case Position.topCenter:
      fAlignProps = "justify-content:center;";
      dAlignProps += "margin-bottom: auto;";
      break;
    case Position.bottomCenter:
      fAlignProps = "justify-content:center;";
      dAlignProps += "margin-top: auto;";
      break;
    case Position.center:
      fAlignProps = "justify-content:center;";
      dAlignProps += "margin-bottom: auto;";
      dAlignProps += "margin-top: auto;";
      break;
  }
  return "\n           <div class='container' style='\n                                  "
    .concat(
      fAlignProps,
      "\n                                  background-color: "
    )
    .concat(BACKGROUND_COLOR, ";\n                                  width: ")
    .concat(
      container.width || "auto",
      ";\n                                  height: "
    )
    .concat(container.height || "auto", ";\n                                  ")
    .concat(
      container.styles || "",
      ";\n            '> \n              <div style='"
    )
    .concat(dAlignProps, "'> ")
    .concat(container.contents, "</div>\n          </div>  \n    ");
};
exports.getContainer = getContainer;
