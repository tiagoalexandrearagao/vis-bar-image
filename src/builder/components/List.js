Object.defineProperty(exports, "__esModule", { value: true });
exports.getList = void 0;
var Common_1 = require("./Common");
var getList = function (list) {
  var listHtml = [];
  var CONFIG = getConfiguration(list);
  listHtml.push(
    " <div class='listCard "
      .concat(
        CONFIG.ShadowClass,
        "' \n                        style='\n                                "
      )
      .concat(
        (0, Common_1.getCommonProps)(list),
        ";\n                                "
      )
      .concat(CONFIG.ListBackgroundColor, ";\n                                ")
      .concat(CONFIG.SideLine, ";\n                                ")
      .concat(
        list.cardStyles,
        ";\n                                '\n                    > "
      )
  );
  listHtml.push(
    "<div class = 'listTitle' \n                        style=' "
      .concat(list.listTitleStyles || "", ";\n                                ")
      .concat(
        list.listTitleStyles,
        ";' \n                        >\n                     "
      )
      .concat(list.title, "\n                    </div> ")
  );
  for (var i = 0; i < list.data.length; i++) {
    listHtml.push(
      "<div class='listItems'\n                          style=' "
        .concat(CONFIG.SeperatorLine, ";\n                                  ")
        .concat(
          list.listItemStyles,
          ";\n                                '\n                      >\n                     "
        )
        .concat(list.data[i], "\n                    </div>")
    );
  }
  listHtml.push("</div>");
  return listHtml.join("");
};
exports.getList = getList;
var getConfiguration = function (list) {
  var LIST_BACKGROUND_COLOR =
    list.backgroundColor != null
      ? "background-color: ".concat(
          (0, Common_1.getThemeShades)(list.backgroundColor).l1
        )
      : "background-color: #fff";
  var SIDE_LINE =
    list.sideLineColor != null
      ? "border-left: 2px solid ".concat(
          (0, Common_1.getThemeShades)(list.sideLineColor).l3
        )
      : "";
  var SHADOW_CLASS = list.hasShadow ? "shadow" : "";
  var SEPERATOR_LINE = "";
  if (list.hasSeparatorLines) {
    SEPERATOR_LINE =
      list.backgroundColor != null
        ? " border-bottom: 1px solid ".concat(
            (0, Common_1.getThemeShades)(list.backgroundColor).l2
          )
        : "";
  }
  var CONFIG = {
    ListBackgroundColor: LIST_BACKGROUND_COLOR,
    SeperatorLine: SEPERATOR_LINE,
    ShadowClass: SHADOW_CLASS,
    SideLine: SIDE_LINE,
  };
  return CONFIG;
};
