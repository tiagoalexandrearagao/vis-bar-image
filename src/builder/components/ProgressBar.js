Object.defineProperty(exports, "__esModule", { value: true });
exports.getProgressBar = void 0;
var Common_1 = require("./Common");
var getProgressBar = function (bar) {
  var themeShades = (0, Common_1.getThemeShades)(bar.color);
  var BAR_HEIGHT = bar.itemsHorizontal ? "4px" : "100%";
  var BAR_WIDTH = bar.itemsHorizontal ? "100%" : "4px";
  var FILL_HEIGHT = bar.itemsHorizontal
    ? "4px"
    : "".concat(bar.percentFull, "%");
  var FILL_WIDTH = bar.itemsHorizontal
    ? "".concat(bar.percentFull, "%")
    : "4px";
  return "\n              <div class='bar' style=' \n                                      background-color:"
    .concat(themeShades.l1, " ;\n                                      width: ")
    .concat(BAR_WIDTH, "; \n                                      height: ")
    .concat(
      BAR_HEIGHT,
      ";\n                                        '> \n                 <div class='barFill' style='\n                                      background-color: "
    )
    .concat(themeShades.l3, ";\n                                      width:")
    .concat(FILL_WIDTH, "; \n                                      height: ")
    .concat(
      FILL_HEIGHT,
      "; '>\n                 </div>\n              </div>\n    "
    );
};
exports.getProgressBar = getProgressBar;
