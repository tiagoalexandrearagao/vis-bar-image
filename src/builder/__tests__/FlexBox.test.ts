import { FlexBox } from '..';
import * as components from '../components';
import { TextType } from '../components';
import { FLEX_CONFIGS } from './Configs/FlexConfigs';

const HEADINGS = ['Name', 'Year', 'scores '];
const DATA = [['Dave', '10', '90'], ['Sophie', '9', '85'], ['Ronaldo', '12', '45']];
const TITLE = 'Student Details';

describe('Flex Test', () => {
  FLEX_CONFIGS.forEach((fProps, index) => {
    it(`Flex Config ${index} `, () => {
      let flex = new FlexBox(fProps);

      flex.add(
        components.getTable({
          data: DATA,
          headings: HEADINGS,
          title: TITLE
        })
      );

      flex.add(
        components.getTable({
          data: DATA,
          headings: HEADINGS,
          title: TITLE
        })
      );
      flex.addDivider();
      flex.add(
        components.getText({
          text: 'This page is for testing',
          type: TextType.h3
        })
      );

      expect(flex.construct().replace(/\n*\t*\r*\s+/g, ' ')).toMatchSnapshot();
    });
  });
});
