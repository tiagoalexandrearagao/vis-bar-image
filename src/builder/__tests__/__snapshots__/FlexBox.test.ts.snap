// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex Test Flex Config 0  1`] = `"<div class='flex' style= ' display: flex;flex-wrap: wrap;flex-direction: column;justify-content: center;width: auto;height: auto;;; '> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div><div class='divider' style='width:auto; height:1px; display:block; background-color: rgba(0,0,0,.12); margin:4px; '> </div><div class='text' style= 'font-size: 20px; color: #000; ; '> This page is for testing </div> </div>"`;

exports[`Flex Test Flex Config 1  1`] = `"<div class='flex' style= ' display: flex;flex-wrap: wrap;flex-direction: row;justify-content: flex-start;width: auto;height: auto;background-color: #BBDEFB;; '> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div><div class='divider' style='width:1px; height:auto; display:block; background-color: rgba(0,0,0,.12); margin:4px; '> </div><div class='text' style= 'font-size: 20px; color: #000; ; '> This page is for testing </div> </div>"`;

exports[`Flex Test Flex Config 2  1`] = `"<div class='flex' style= ' display: flex;flex-wrap: wrap;flex-direction: row;justify-content: flex-end;width: auto;height: auto;background-color: #F5F5F5;; '> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div><div class='divider' style='width:1px; height:auto; display:block; background-color: rgba(0,0,0,.12); margin:4px; '> </div><div class='text' style= 'font-size: 20px; color: #000; ; '> This page is for testing </div> </div>"`;

exports[`Flex Test Flex Config 3  1`] = `"<div class='flex' style= ' display: flex;flex-wrap: wrap;flex-direction: row;justify-content: space-around;width: auto;height: auto;;; '> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div><div class='divider' style='width:1px; height:auto; display:block; background-color: rgba(0,0,0,.12); margin:4px; '> </div><div class='text' style= 'font-size: 20px; color: #000; ; '> This page is for testing </div> </div>"`;

exports[`Flex Test Flex Config 4  1`] = `"<div class='flex' style= ' display: flex;flex-wrap: wrap;flex-direction: row;justify-content: space-between;width: auto;height: auto;;; '> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div><div class='divider' style='width:1px; height:auto; display:block; background-color: rgba(0,0,0,.12); margin:4px; '> </div><div class='text' style= 'font-size: 20px; color: #000; ; '> This page is for testing </div> </div>"`;

exports[`Flex Test Flex Config 5  1`] = `"<div class='flex' style= ' display: flex;flex-wrap: wrap;flex-direction: row;justify-content: space-evenly;width: auto;height: auto;;; '> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div><div class='divider' style='width:1px; height:auto; display:block; background-color: rgba(0,0,0,.12); margin:4px; '> </div><div class='text' style= 'font-size: 20px; color: #000; ; '> This page is for testing </div> </div>"`;

exports[`Flex Test Flex Config 6  1`] = `"<div class='flex' style= ' display: flex;flex-wrap: wrap;flex-direction: row;justify-content: stretch;width: auto;height: auto;;; '> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div> <div class ='tableCard ' style='width: fit-content;height: auto;margin: 4px 4px; ; ;' > <div class='tableTitle' style=' text-align: left; '> Student Details </div> <table class='table ' style='; ;'> <thead class='tableHead' style=' background-color: #000; ; '> <th class='tableHeadings' style='text-align: left; '> Name </th><th class='tableHeadings' style='text-align: left; '> Year </th><th class='tableHeadings' style='text-align: left; '> scores </th></thead><tbody><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Dave </td><td class='tableCell' style= '; text-align: left; '> 10 </td><td class='tableCell' style= '; text-align: left; '> 90 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Sophie </td><td class='tableCell' style= '; text-align: left; '> 9 </td><td class='tableCell' style= '; text-align: left; '> 85 </td></tr><tr class ='tableRowOdd' style=';'><td class='tableCell' style= '; text-align: left; '> Ronaldo </td><td class='tableCell' style= '; text-align: left; '> 12 </td><td class='tableCell' style= '; text-align: left; '> 45 </td></tr></tbody></table></div><div class='divider' style='width:1px; height:auto; display:block; background-color: rgba(0,0,0,.12); margin:4px; '> </div><div class='text' style= 'font-size: 20px; color: #000; ; '> This page is for testing </div> </div>"`;
